src/python/pants/engine/fs.py

```diff
@@ -9,7 +9,7 @@
 from pants.base.project_tree import Dir, File
 from pants.engine.rules import RootRule
 from pants.option.global_options import GlobMatchErrorBehavior
-from pants.util.objects import Collection, datatype
+from pants.util.objects import Collection, datatype, enum
```

Comment:
IMO, the enum bit is out of scope for a bugfix like this. Would be good to do this with a simple fix, and then discuss adding more datatype features later.

Reply:
I specifically noted that the `enum` method should be considered experimental and unstable in f6172a9!